app/src/main/java/woowacourse/shopping/presentation/goods/list/GoodsViewModel.kt

```diff
@@ -1,52 +1,187 @@
 package woowacourse.shopping.presentation.goods.list
 
+import android.util.Log
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.viewmodel.initializer
 import androidx.lifecycle.viewmodel.viewModelFactory
-import woowacourse.shopping.data.GoodsRepository
 import woowacourse.shopping.domain.model.Goods
+import woowacourse.shopping.domain.model.ShoppingGoods
+import woowacourse.shopping.domain.repository.GoodsRepository
+import woowacourse.shopping.domain.repository.LatestGoodsRepository
+import woowacourse.shopping.domain.repository.ShoppingRepository
+import woowacourse.shopping.presentation.util.event.MutableSingleLiveData
+import woowacourse.shopping.presentation.util.event.SingleLiveData
 
 class GoodsViewModel(
     private val goodsRepository: GoodsRepository,
+    private val shoppingRepository: ShoppingRepository,
+    private val latestGoodsRepository: LatestGoodsRepository,
 ) : ViewModel() {
     private val _goods: MutableLiveData<List<Goods>> = MutableLiveData()
     val goods: LiveData<List<Goods>>
         get() = _goods
 
+    private val _shoppingGoodsCount: MutableLiveData<Int> = MutableLiveData(MINIMUM_QUANTITY)
+    val shoppingGoodsCount: LiveData<Int>
+        get() = _shoppingGoodsCount
+
+    private val _latestGoods: MutableLiveData<List<Goods>> = MutableLiveData()
+    val latestGoods: LiveData<List<Goods>>
+        get() = _latestGoods
+
     private val _shouldShowLoadMore: MutableLiveData<Boolean> = MutableLiveData(false)
     val shouldShowLoadMore: LiveData<Boolean>
         get() = _shouldShowLoadMore
 
+    private val _shouldNavigateToShoppingCart: MutableSingleLiveData<Unit> = MutableSingleLiveData()
+    val shouldNavigateToShoppingCart: SingleLiveData<Unit>
+        get() = _shouldNavigateToShoppingCart
+
+    private val _onQuantityChanged: MutableSingleLiveData<Int> = MutableSingleLiveData()
+    val onQuantityChanged: SingleLiveData<Int>
+        get() = _onQuantityChanged
+
     private var page: Int = DEFAULT_PAGE
 
-    init {
-        _goods.value = goodsRepository.getPagedGoods(page++, ITEM_COUNT)
+    fun initGoods() {
+        goodsRepository.getPagedGoods(page, ITEM_COUNT) { currentGoods ->
+            shoppingRepository.getAllGoods { selectedItems ->
+                val updatedGoods = getUpdatedGoods(currentGoods, selectedItems)
+
+                _goods.postValue(updatedGoods)
```

Comment:
```suggestion
                _goods.value = updatedGoods
```

이런 형태와 `postValue`는 어떤 차이가 있을까요?

만약 차이를 `setValue`를 하면 터져서 찾아보니 `postValue` 로 바꾼 것이라면, 왜 `setValue` 는 안되는지를 고민해 보시기 바랍니다. 비동기 / 멀티 쓰레드 환경에서는 항상 중요하게 여겨야 하는 포인트입니다.

Reply:
`setValue`는 `MainThread`에서, `postValue`는 `백그라운드 Thread`에서 작동합니다

 해당 콜백 함수를 호출하는 부분이 `thread` 안에 있기 때문에,
`setValue`를 사용할 시 백그라운드 호출로 에러가 생기는 것 같습니다 ... !